ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp

```diff
@@ -218,6 +219,29 @@ Tensor BinaryOperation<binary_op_type, in_place>::operator()(
 }
 
 
+template <BinaryOpType binary_op_type, bool in_place>
+ComplexTensor BinaryOperation<binary_op_type, in_place>::operator()(
+    const ComplexTensor &input_a,
+    const ComplexTensor &input_b,
+    const ttnn::MemoryConfig &output_mem_config) {
+    TT_FATAL((binary_op_type == BinaryOpType::MUL || binary_op_type == BinaryOpType::DIV_FAST) && "Dimension out of range (expected to be in range of [-4, 3]");
```

Comment:
Would prefer for `TT_FATAL` to be in an `else` branch, and at that point it can also be unconditional. The point is that you don't need to check this at runtime when `binary_op_type` is known at compile-time. Should also allow you to uncomment the `else if constexpr` below if you put this in `else` afterwards.